x/cronos/middleware/conversion_middleware.go

```diff
@@ -213,11 +213,20 @@ func (im IBCConversionModule) canBeConverted(ctx sdk.Context, denom string) bool
 
 func (im IBCConversionModule) getIbcDenomFromPacketAndData(
 	packet channeltypes.Packet, data transferTypes.FungibleTokenPacketData,
+) string {
+	return im.getIbcDenomFromPacketAndDataBySource(packet, data, false)
+}
+
+func (im IBCConversionModule) getIbcDenomFromPacketAndDataBySource(
```

Comment:
I was thinking something like that 

```
func (im IBCConversionModule) getIbcDenomFromPacketAndData(
	packet channeltypes.Packet, data transferTypes.FungibleTokenPacketData, refund bool
) string {


	isSource := transferTypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), data.Denom) 

	if refund {
		isSource = !isSource
	}	

	if source {
		voucherPrefix := transferTypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
		unprefixedDenom := data.Denom[len(voucherPrefix):]
		denom := unprefixedDenom
		denomTrace := transferTypes.ParseDenomTrace(unprefixedDenom)
		if denomTrace.Path != "" {
			denom = denomTrace.IBCDenom()
		}
		return denom
	}

	// since SendPacket did not prefix the denomination, we must prefix denomination here
	sourcePrefix := transferTypes.GetDenomPrefix(packet.GetDestPort(), packet.GetDestChannel())
	prefixedDenom := sourcePrefix + data.Denom
	denomTrace := transferTypes.ParseDenomTrace(prefixedDenom)
	return denomTrace.IBCDenom()
}
```

could you give a try?